test_runner/regress/test_gc_old_layers.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder
+
+
+# Test gc_old_layers
+#
+# This test sets fail point at the end of GC, and checks that pageserver
+# normally restarts after it. Also, there should be GC ERRORs in the log,
+# but the fixture checks the log for any unexpected ERRORs after every
+# test anyway, so it doesn't need any special attention here.
+@pytest.mark.timeout(10000)
+def test_gc_old_layers(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+    client = env.pageserver.http_client()
+
+    # Use aggressive GC and checkpoint settings, so that we also exercise GC during the test
+    tenant_id, _ = env.neon_cli.create_tenant(
+        conf={
+            # disable default GC and compaction
+            "gc_period": "1000 m",
```

Comment:
```suggestion
            "gc_period": "0 s",
```

Reply:
As I've found recently, this doesn't really affect this test because it doesn't do a restart or detach+attach, it would be surprising that the gc is ran on re-start if this is non-zero.

Reply:
Aah ok, yeah it annoys me as well. I'll PR a change to make it less loggy but still check it every 10s?

Reply:
Hmmm...
Actually I do not know right answer but few notes:
1. With PiTR equal to one or few days, there is not so much sense to perform GC more frequently than once per day.
2. Disabling automatic GC may be needed only for testing. I do not see any reasons to disable automatic GC temporary or permanently at production or staging
3. For tests it may be useful to use less poll interval - for example 1 second. Right now we do not have test which disables + enables automatic GC But there will be such tests than shorter GC chek interval is preferrable.

May be the best solution is to prohibit 0 value at all. But then some tests needs to be changed.

Reply:
> With PiTR equal to one or few days, there is not so much sense to perform GC more frequently than once per day.

I cannot see how this would be true. Suppose there are heavy updates most of the time, then wouldn't some "less than a day" increment be able to free some layers often? Not saying that the once per pitr_period wouldn't be able to free up some space, just saying it would be able to do it much earlier as well.

> Disabling automatic GC may be needed only for testing. I do not see any reasons to disable automatic GC temporary or permanently at production or staging

Yeah we mostly need it for testing, which is why I'd like to keep a single way of disabling it in testing; some flakyness has been caused by surprise gc at the restart with high gc_period.

> For tests it may be useful to use less poll interval - for example 1 second. Right now we do not have test which disables + enables automatic GC But there will be such tests than shorter GC chek interval is preferrable.

Well, not really related response, but inspired by it: I think we should just quit the sleep whenever the tenant settings are updated, however there are tech debt cleanups piled up on these background tasks, for example: #3501.

Reply:
I will not get the PR done this week, created an issue.

I think this thread could now be resolved, be it `1000 min` or `0s`.